src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const sepatareDate = date.split(fromFormat[3]);
+  const newDate = [];
+  const oldYearIndex = fromFormat.indexOf('YY');
+  const separator = toFormat[3];
+
+  if (sepatareDate[oldYearIndex] >= 30) {
+    sepatareDate[oldYearIndex] = '19' + sepatareDate[oldYearIndex];
+  }
+
+  if (sepatareDate[oldYearIndex] < 30) {
+    sepatareDate[oldYearIndex] = '20' + sepatareDate[oldYearIndex];
+  }
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
```

Comment:
The loop should iterate over `fromFormat` excluding the separator, which is correctly done here. However, ensure that the logic inside the loop correctly handles both 'YY' and 'YYYY' conversions.